src/Access/Common/AccessType.h

```diff
@@ -203,6 +203,44 @@ enum class AccessType
     M(demangle, "", GLOBAL, INTROSPECTION) /* allows to execute function demangle() */\
     M(INTROSPECTION, "INTROSPECTION FUNCTIONS", GROUP, ALL) /* allows to execute functions addressToLine(), addressToSymbol(), demangle()*/\
     \
+    M(MergeTree, "", GLOBAL, SOURCES) \
```

Comment:
That doesn't look completely correct. I mean `URL` or `AZURE` are sources, but for example `CollapsingMergeTree` is not a source, it's a table engine.

Perhaps it's better to invent another privilege group `ALL_TABLE_ENGINES` and make `SOURCES` a subset of it. Also it would be nice to group similar table engines into groups according to the [engine families](https://clickhouse.com/docs/en/engines/table-engines) like `MERGE_TREE_TABLE_ENGINES` and `LOG_TABLE_ENGINES`, i.e.
```
M(MergeTree, "", GLOBAL, MERGE_TREE_TABLE_ENGINES) \
M(CollapsingMergeTree, "", GLOBAL, MERGE_TREE_TABLE_ENGINES) \
...
M(Log, "", GLOBAL, LOG_TABLE_ENGINES) \
M(TinyLog, "", GLOBAL, LOG_TABLE_ENGINES) \
M(StripeLog, "", GLOBAL, LOG_TABLE_ENGINES) \
...
M(Buffer, GLOBAL, SPECIAL_TABLE_ENGINES) \
...
M(S3, "", GLOBAL, INTEGRATION_TABLE_ENGINES) \
...
M(MERGE_TREE_TABLE_ENGINES, "", GROUP, ALL_TABLE_ENGINES) \
M(LOG_TABLE_ENGINES, "", GROUP, ALL_TABLE_ENGINES) \
M(SPECIAL_TABLE_ENGINES, "", GROUP, ALL_TABLE_ENGINES) \
M(INTEGRATION_TABLE_ENGINES, "SOURCES", GROUP, ALL_TABLE_ENGINES) \
M(ALL_TABLE_ENGINES, "", GROUP, ALL) \
```

We can turn `SOURCES` into an alias of `INTEGRATION_TABLE_ENGINES`, that seems quite right.

Reply:
cannot agree more, would do that.